app/src/main/java/ch/epfl/sdp/appart/ad/UserAdsViewModel.java

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.sdp.appart.ad;
+
+import android.util.Log;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import ch.epfl.sdp.appart.database.DatabaseService;
+import ch.epfl.sdp.appart.login.LoginService;
+import ch.epfl.sdp.appart.scrolling.card.Card;
+import ch.epfl.sdp.appart.user.User;
+import dagger.hilt.android.lifecycle.HiltViewModel;
+
+@HiltViewModel
+public class UserAdsViewModel extends ViewModel {
+    private final MutableLiveData<List<Card>> lUserAds = new MutableLiveData<>();
+    final DatabaseService database;
+    final LoginService loginService;
+
+    @Inject
+    public UserAdsViewModel(DatabaseService database, LoginService loginService) {
+        this.database = database;
+        this.loginService = loginService;
+    }
+
+    public void initHome() {

```

Comment:
Consider making initHome return a completable future to tell the activity whether the initialization was successful. It will save us a possible crash in case of issues with the database. Maybe in case of error you could also set the observable list with an empty list

Reply:
Thing is, I did it like that to stay coherent with ScrollingActivity and FavoritesActivity, which implement a similar system of ViewModel. I agree that doing as you said would be way more robust, but I think we should refactor it later along with the others, so the project stays coherent.